libraries/chain/controller.cpp

```diff
@@ -85,6 +92,9 @@ struct controller_impl {
     db( cfg.shared_memory_dir,
         cfg.read_only ? database::read_only : database::read_write,
         cfg.shared_memory_size ),
+    unconfirmed_blocks( cfg.block_log_dir/"unconfirmed",
+        cfg.read_only ? database::read_only : database::read_write,
+        std::min<uint64_t>(cfg.shared_memory_size,128*1024*1024ll) ), /// 1GB should store 1000 1MB blocks + overhead
```

Comment:
The expected number of unconfirmed blocks in a health network of 21 producers is around 360. With 1 MB blocks, that would mean we would need at least 360 MB for the unconfirmed database, and this line appears to put a hard ceiling of 128 MB.

In an unhealthy network, the number of blocks required could easily exceed 360 and if the unconfirmed database size is too small, it will compound the problem by taking down more nodes (including potentially block producing nodes).

I think this should be a separate configuration parameter like `cfg.shared_memory_size` and the default value should be at least 512 MB.